be/src/exec/rowid_fetcher.cpp

```diff
@@ -717,19 +749,226 @@ Status RowIdStorageReader::read_batch_external_row(const PRequestBlockDesc& requ
+                                                    file_read_bytes_counter->value(),
+                                                    file_read_bytes_counter->type());
+                                        }
+
+                                        auto file_read_times_counter =
+                                                sub_runtime_profile->get_counter(
+                                                        vectorized::FileScanner::
+                                                                FileReadTimeProfile);
+                                        if (file_read_times_counter != nullptr) {
+                                            file_read_times[idx] = PrettyPrinter::print(
+                                                    file_read_times_counter->value(),
+                                                    file_read_times_counter->type());
+                                        }
+
+                                        semaphore.release();
+                                        if (++producer_count == scan_rows.size()) {
+                                            std::lock_guard<std::mutex> lock(mtx);
+                                            cv.notify_one();
+                                        }
+                                        return Status::OK();
+                                    },
+                                    nullptr)));
+                    idx++;
+                }
 
-        // Clear to avoid reading iceberg position delete file...
-        scan_range_desc.table_format_params.iceberg_params = TIcebergFileDesc {};
+                {
+                    std::unique_lock<std::mutex> lock(mtx);
+                    cv.wait(lock, [&] { return producer_count == scan_rows.size(); });
+                }
+                return Status::OK();
+            },
+            &scan_running_time));
+
+    // Insert the read data into result_block.
+    for (const auto& [pos_block, block_idx] : row_id_block_idx) {
+        for (size_t column_id = 0; column_id < result_block.get_columns().size(); column_id++) {
+            auto dst_col =
+                    const_cast<vectorized::IColumn*>(result_block.get_columns()[column_id].get());
 
-        // Clear to avoid reading hive transactional delete delta file...
-        scan_range_desc.table_format_params.transactional_hive_params = TTransactionalHiveDesc {};
+            DCHECK(scan_blocks.size() > pos_block);
+            DCHECK(scan_blocks[pos_block].get_columns().size() > column_id);
+            auto& src_col = *scan_blocks[pos_block].get_columns()[column_id].get();
 
-        RETURN_IF_ERROR(vfile_scanner_ptr->read_one_line_from_range(
-                scan_range_desc, request_block_desc.row_id(j), &result_block, external_info,
-                init_reader_ms, get_block_ms));
+            dst_col->insert_range_from(src_col, block_idx, 1);
```

Comment:
length 1, can use `insert_from` to do the work。
if want speed up the call， also can call `insert_from_multi_column`